src/components/RecipientCard.vue

```diff
@@ -1,58 +1,160 @@
 <template>
     <SmallPage class="recipient-card">
-        <Identicon :address="address"></Identicon>
-        <AddressDisplay :address="address"></AddressDisplay>
-        <hr class="separator" />
-        <input v-model="message" class="nq-input vanishing" type="text" placeholder="Add a Message"/>
-        <div class="nq-text-s">(optional)</div>
+        <transition name="transition-fade">
+            <div v-if="!isAddressValid">
+                <h1 class="nq-h1">Enter an Address</h1>
+                <p class="nq-text">All received donations will be sent to this Address.</p>
+                <div class="address-input-wrapper">
+                    <AddressInput v-model="address"/>
+                </div>
+                <button class="nq-button-s" @click="chooseAddress">Login existing Account</button>
+            </div>
+            <div v-else>
+                <ArrowLeftIcon @click="goBack" />
+                <Identicon :address="address"></Identicon>
+                <AddressDisplay :address="address"></AddressDisplay>
+                <hr class="separator" />
+                <input v-model="message" class="nq-input vanishing" type="text" placeholder="Add a message..."/>
+                <div class="nq-text-s">(optional)</div>
+            </div>
+        </transition>
     </SmallPage>
 </template>
 
-
 <script lang="ts">
-import { SmallPage, Identicon, AddressDisplay } from '@nimiq/vue-components';
-import { Component, Vue, Prop } from 'vue-property-decorator';
+import {
+    SmallPage,
+    AddressInput,
+    AddressDisplay,
+    Identicon,
+    ArrowLeftIcon,
+} from '@nimiq/vue-components';
+import { Component, Vue, Prop, Watch } from 'vue-property-decorator';
+import { ValidationUtils } from '@nimiq/utils';
+import HubApi from '@nimiq/hub-api';
 
-@Component({components: {SmallPage, Identicon, AddressDisplay}})
+@Component({ components: {
+    SmallPage,
+    AddressInput,
+    AddressDisplay,
+    Identicon,
+    ArrowLeftIcon,
+}})
 export default class RecipientCard extends Vue {
-
-    @Prop(String) public address!: string;
+    private static hubApi: any = new HubApi('https://hub.nimiq.com');
```

Comment:
> Interesting that you need to type this as any. @nimiq/hub-api should provide types, doesn't it?

Actually i don't need to type it as `any` so i'll remove it. But i need to declare the module in the `types` folder because the hub-api is not typed apparently.

> Note that usually in the testnet we also use the testnet hub, but maybe here in the donation app it actually makes more sense to enforce the mainnet hub, so that's fine.

In fact, i was planning to select the url based on the `buildName` in the `vue.config.js` file. But since it was not making so much sense for the donation app i just enforced the mainnet.